lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint.py

```diff
@@ -0,0 +1,384 @@
+    default: no
+    choices: ["yes", "no"]
+  wait_timeout:
+    description:
+      - Used in conjunction with wait. Number of seconds to wait for status.
+        Unfortunately this is ignored for delete actions due to a difference in
+        behaviour from AWS.
+    required: false
+    default: 320
+  route_table_ids:
+    description:
+      - List of one or more route table ids to attach to the endpoint. A route
+        is added to the route table with the destination of the endpoint if
+        provided.
+    required: false
+  vpc_endpoint_id:
+    description:
+      - One or more vpc endpoint ids to remove from the AWS account
+    required: false
+  client_token:
+    description:
+      - Optional client token to ensure idempotency
+    required: false
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+- name: Create new vpc endpoint with a json template for policy
+  ec2_vpc_endpoint:
+    state: present
+    region: ap-southeast-2
+    vpc_id: vpc-12345678
+    service: com.amazonaws.ap-southeast-2.s3
+    policy: " {{ lookup( 'template', 'endpoint_policy.json.j2') }} "
+    route_table_ids:
+      - rtb-12345678
+      - rtb-87654321
+  register: new_vpc_endpoint
+
+- name: Create new vpc endpoint the default policy
+  ec2_vpc_endpoint:
+    state: present
+    region: ap-southeast-2
+    vpc_id: vpc-12345678
+    service: com.amazonaws.ap-southeast-2.s3
+    route_table_ids:
+      - rtb-12345678
+      - rtb-87654321
+  register: new_vpc_endpoint
+
+- name: Create new vpc endpoint with json file
+  ec2_vpc_endpoint:
+    state: present
+    region: ap-southeast-2
+    vpc_id: vpc-12345678
+    service: com.amazonaws.ap-southeast-2.s3
+    policy_file: "{{ role_path }}/files/endpoint_policy.json"
+    route_table_ids:
+      - rtb-12345678
+      - rtb-87654321
+  register: new_vpc_endpoint
+
+- name: Delete newly created vpc endpoint
+  ec2_vpc_endpoint:
+    state: absent
+    nat_gateway_id: "{{ new_vpc_endpoint.result['VpcEndpointId'] }}"
+    region: ap-southeast-2
+'''
+
+RETURN = '''
+endpoints:
+  description: The resulting endpoints from the module call
+  returned: success
+  type: list
+  sample: [
+      {
+        "creation_timestamp": "2017-02-20T05:04:15+00:00",
+        "policy_document": {
+          "Id": "Policy1450910922815",
+          "Statement": [
+            {
+              "Action": "s3:*",
+              "Effect": "Allow",
+              "Principal": "*",
+              "Resource": [
+                "arn:aws:s3:::*/*",
+                "arn:aws:s3:::*"
+              ],
+              "Sid": "Stmt1450910920641"
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "route_table_ids": [
+          "rtb-abcd1234"
+        ],
+        "service_name": "com.amazonaws.ap-southeast-2.s3",
+        "vpc_endpoint_id": "vpce-a1b2c3d4",
+        "vpc_id": "vpc-abbad0d0"
+      }
+    ]
+'''
+
+try:
+    import botocore
```

Comment:
I would look for boto3 specifically.  You can have botocore without boto3 but not vice versa